src/micro_http/src/server.rs

```diff
@@ -300,10 +300,11 @@ impl HttpServer {
         // current thread until at least one event is received.
         // The received notifications will then populate the `events` array with
         // `event_count` elements, where 1 <= event_count <= MAX_CONNECTIONS.
-        let event_count = self
-            .epoll
-            .wait(MAX_CONNECTIONS, -1, &mut events[..])
-            .map_err(ServerError::IOError)?;
+        let event_count = match self.epoll.wait(MAX_CONNECTIONS, -1, &mut events[..]) {
+            Ok(event_count) => event_count,
+            Err(e) if e.raw_os_error() == Some(libc::EINTR) => 0,
```

Comment:
We can directly return `parsed_requests` here.

Reply:
Since this is a very unlikely code path, I prefer to keep it simple at a very small expense in efficiency.

In terms of separation of concerns, the logic in this code block is currently contained to epoll-related logic. There is _a lot_ of code happening after this code-block. To short-circuit the function here would mean to introduce an assumption in this code block that all following code is irrelevant in an EINTR case (which is true right now, but adds validation burden to anyone making changes here in the future).

I will reluctantly short-circuit the function if you have a strong opinion we should do it. Let me know.